aicsimageio/readers/lif_reader.py

```diff
@@ -0,0 +1,767 @@
+        # of the array at the same multi-index to the delayed reader using the constructed read_dims dictionary.
+        dims = [d for d in Dimensions.DefaultOrder]
+        begin_indicies = tuple(image_dim_indices[d][0] for d in dims)
+        for i, _ in np.ndenumerate(lazy_arrays):
+            # Add the czi file begin index for each dimension to the array dimension index
+            this_chunk_read_indicies = (
+                current_dim_begin_index + curr_dim_index
+                for current_dim_begin_index, curr_dim_index in zip(begin_indicies, i)
+            )
+
+            # Zip the dims with the read indices
+            this_chunk_read_dims = dict(zip(blocked_dimension_order, this_chunk_read_indicies))
+
+            # Remove the dimensions that we want to chunk by from the read dims
+            for d in chunk_by_dims:
+                if d in this_chunk_read_dims:
+                    this_chunk_read_dims.pop(d)
+
+            # Add delayed array to lazy arrays at index
+            lazy_arrays[i] = da.from_delayed(
+                delayed(LifReader._imread)(img, offsets, read_length, lif.xml_root, this_chunk_read_dims),
+                shape=sample_chunk_shape,
+                dtype=sample.dtype,
+            )
+
+        # Convert the numpy array of lazy readers into a dask array and fill the inner-most empty dimensions with chunks
+        merged = da.block(lazy_arrays.tolist())
+
+        # Because we have set certain dimensions to be chunked and others not
+        # we will need to transpose back to original dimension ordering
+        # Example being, if the original dimension ordering was "SZYX" and we want to chunk by "S", "Y", and "X"
+        # We created an array with dimensions ordering "ZSYX"
+        transpose_indices = []
+        transpose_required = False
+        for i, d in enumerate(Dimensions.DefaultOrder):
+            new_index = blocked_dimension_order.index(d)
+            if new_index != i:
+                transpose_required = True
+                transpose_indices.append(new_index)
+            else:
+                transpose_indices.append(i)
+
+        # Only run if the transpose is actually required
+        # The default case is "Z", "Y", "X", which _usually_ doesn't need to be transposed because that is _usually_
+        # The normal dimension order of the CZI file anyway
+        if transpose_required:
+            merged = da.transpose(merged, tuple(transpose_indices))
+
+        # Because dimensions outside of Y and X can be in any order and present or not
+        # we also return the dimension order string.
+        return merged, "".join(dims)
+
+    @property
+    def dask_data(self) -> da.core.Array:
+        """
+        Returns
+        -------
+        Constructed dask array where each chunk is a delayed read from the CZI file.
+        Places dimensions in the native order (i.e. "TZCYX")
```

Comment:
`TZCYX` is the normal ordering? Weird.

Reply:
I'd have to go through the offset calculations again but I believe that is how they are recorded in the file.